LukeZvada/flashcards-server
# Categories

# Description

- Adds a few categories in fixtures to get us started
- All users can view categories
- All users can view a single category
- All users can search for a category
- Only admins can create a new category
- Only admins can delete a category
- Only admins can edit a category
- Only admins can approve a category

Fixes # (issue)
closes #15 
closed #23
## Type of change

# How Has This Been Tested?

- Run your reset.sh bash script to load users, tokens, etc

- ` e48584ea29993d6e7b6fc97da6e29ab1c24ef1e7 `  -- Admin User

- `079ef37acfa281ee45cbebb5199728940b4b4194 `   -- Regular User

1.  GET request in Postman `http://localhost:8000/categories` with either token should display all categories

2.  DELETE request in Postman `http://localhost:8000/categories/1` with Admin user should delete that category.
The same request the regular user gives a 403 error

3. POST request in Postman `http://localhost:8000/categories` with Admin users with `{"label" : "A new Label"}`
in the body should give a 201 status. When fetched that category will show as 'approved' : false
The same request with a regular user gives a 403 error

4. PUT request in Postman `http://localhost:8000/categories/2` with Admin user with `{"label" : "Edited Label"} ` in the body
should give a 204 status. When edited, a category will automatically change to 'approved' : false. 
The same request with a regular user gives a 403 error

5.  Custom Action
PATCH request in Postman `http://localhost:8000/categories/3/approve` with an Admin user with `{"approved" : true}` should give a 204 status and change that category to approved.
The same request with a regular user gives a 403 error.



# Checklist:
